scripts/wsrep_sst_clone.sh

```diff
@@ -0,0 +1,1251 @@
+            else
+                CLIENT_SSL_OPTIONS+=" --ssl-mode=REQUIRED"
+            fi
+        fi
+    else
+        wsrep_log_info "No suitable SSL configuration found. Not using SSL for SST."
+        CLEANUP_CLONE_SSL=
+        REQUIRE_SSL=""
+    fi
+    wsrep_log_debug "-> ############## SSL SECTION [END] ($ROLE)############"
+}
+
+# Let's use the simple approach to detect if the transfer is running or not.
+# Monitor p_s.clone_progress table. If anything changes there, it means something
+# is happening.
+# One of the columns is DATA: The amount of data transferred in current state, in bytes,
+# so we should be good.
+# Allow transfer stuck for sst-idle-timeout
+monitor_sst_progress() {
+  local progress_monitor_command=$1
+  local timeout=$2
+  local clone_instance_pid=$3
+  local parent_process_pid=$4
+
+  local current_timeout=0
+  local previous_command_output=
+  local sleep=5
+  local previous_progress=
+
+  wsrep_log_debug "progress_monitor_command: ${progress_monitor_command}"
+  wsrep_log_debug "timeout: ${timeout}"
+  wsrep_log_debug "clone_instance_pid: ${clone_instance_pid}"
+  wsrep_log_debug "parent_process_pid: ${parent_process_pid}"
+
+  if [ ${timeout} -eq 0 ]; then
+    return 2;
+  fi
+
+  while true; do
+    # check if parent process is still alive
+    # Is it needed? If parent process dies, SST script dies as well
+    # and will cleanup clone instance in cleanup_joiner() trap.
+    kill -0 $parent_process_pid 2> /dev/null
+    if [ $? -ne 0 ]; then
+      return 1;
+    fi
+
+    # check if clone instance process is still alive
+    kill -0 $clone_instance_pid 2> /dev/null
+    if [ $? -ne 0 ]; then
+      # Clone instance is done with its job
+      return 0;
+    fi
+
+    # check if there is any SST progress
+    command_output=$(eval "$progress_monitor_command" 2> /dev/null ||:)
+    if [ "${command_output}" != "${previous_command_output}" ]; then
+        current_timeout=0
+        previous_command_output=${command_output}
+        # if the transfer is ongoing, keep checking once per 5 seconds
+        sleep=5
+    else
+        current_timeout=$((current_timeout + 1))
+        wsrep_log_debug "sst idle for ${current_timeout} sec"
+        # if we detected transfer stall, start checking more often (1 sec)
+        sleep=1
+    fi
+
+    if [ ${current_timeout} -eq  ${timeout} ]; then
+      return 2
+    fi
+
+    # Report the current progress
+    current_progress=`$MYSQL_ACLIENT -NB -e "select format(((data/estimate)*100),2) 'completed%' from performance_schema.clone_progress where stage like 'FILE_COPY';" 2> /dev/null` || :
+    if [ "$previous_progress" != "$current_progress" ]; then
+        wsrep_log_info "SST progress: ${current_progress}%"
+        previous_progress=$current_progress
+    fi
+
+    sleep ${sleep};
+  done
+
+  wsrep_log_debug "We should never get here"
+  return 0
+}
+
+wsrep_log_debug "-> In the wsrep_sst_clone "
```

Comment:
This message does not add any value. Let's remove it.
```
2025-02-11T07:45:11.214842Z 0 [Note] [MY-000000] [WSREP-SST] (debug) -> WSREP_SST_OPT_ADDR_LOCAL: 127.0.0.1:13009
2025-02-11T07:45:11.214858Z 0 [Note] [MY-000000] [WSREP-SST] (debug) -> WSREP_SST_OPT_LPORT: 
2025-02-11T07:45:11.215282Z 0 [Note] [MY-000000] [WSREP-SST] (debug) -> In the wsrep_sst_clone 
2025-02-11T07:45:11.232045Z 0 [Note] [MY-000000] [WSREP-SST] (debug) -> WSREP_SST_OPT_PORT 13009
2025-02-11T07:45:11.232068Z 0 [Note] [MY-000000] [WSREP-SST] (debug) -> WSREP_SST_OPT_ADDR_LOCAL 127.0.0.1:13009
```

Reply:
not convinced,  this was useful to understand what transformation was done in common and what in the script, and how they were interacting. 
I would keep it is a debug info at the end

Reply:
I meant to only remove the line that prints `In the wsrep_sst_clone`

Reply:
Done. Added comment that this is the point where the script execution starts.